rpc/session.c

```diff
@@ -868,7 +871,11 @@ M0_INTERNAL void m0_rpc_session_cancel(struct m0_rpc_session *session)
 leave_unlock:
 	m0_rpc_machine_unlock(session->s_conn->c_rpc_machine);
 	M0_POST(pending_item_tlist_is_empty(&session->s_pending_cache));
-	M0_POST(session->s_sm.sm_state == M0_RPC_SESSION_IDLE);
+	if (!M0_IN(session_state(session),
```

Comment:
The same as before: we check the state outside of the group lock.

Reply:
After moving first check inside lock, I think we don't need second check again.